src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/EndBiomeProviderMixin.java

```diff
@@ -17,63 +16,43 @@
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 
-import java.util.List;
-
 @Mixin(EndBiomeProvider.class)
-public abstract class EndBiomeProviderMixin extends BiomeProvider {
-	@Shadow
-	@Final
-	private SimplexNoiseGenerator generator;
+public abstract class EndBiomeProviderMixin {
 	@Shadow
 	@Final
 	private Registry<Biome> lookupRegistry;
 	@Shadow
 	@Final
-	private Biome theEndBiome;
-	@Shadow
-	@Final
-	private Biome endHighlandsBiome;
-	@Shadow
-	@Final
 	private Biome endMidlandsBiome;
 	@Shadow
 	@Final
-	private Biome smallEndIslandsBiome;
+	private Biome endBarrensBiome;
 	@Shadow
 	@Final
-	private Biome endBarrensBiome;
-
+	private Biome endHighlandsBiome;
+	
 	private Layer noiseBiomeLayer;
-
-	private EndBiomeProviderMixin(List<Biome> biomes) {
-		super(biomes);
-	}
-
-	@Inject(at = @At("RETURN"), method = "<init>")
-	private void init(Registry<Biome> lookupRegistry, long seed, CallbackInfo info) {
+	
+	@Inject(at = @At("RETURN"), method = "<init>(Lnet/minecraft/util/registry/Registry;JLnet/minecraft/world/biome/Biome;Lnet/minecraft/world/biome/Biome;Lnet/minecraft/world/biome/Biome;Lnet/minecraft/world/biome/Biome;Lnet/minecraft/world/biome/Biome;)V")
+	private void init(Registry<Biome> lookupRegistry, long seed, Biome theEndBiome, Biome endHighlandsBiome, Biome endMidlandsBiome, Biome smallEndIslandsBiome, Biome endBarrensBiome, CallbackInfo ci) {
 		this.noiseBiomeLayer = ACLayerUtil.createEndBiomeLayer(lookupRegistry, (seedModifier) -> new LazyAreaLayerContext(25, seed, seedModifier));
 	}
-
-	@Inject(at = @At("HEAD"), method = "getNoiseBiome(III)Lnet/minecraft/world/biome/Biome;", cancellable = true)
+	
+	@Inject(at = @At("RETURN"), method = "getNoiseBiome(III)Lnet/minecraft/world/biome/Biome;", cancellable = true)
 	private void addEndBiomes(int x, int y, int z, CallbackInfoReturnable<Biome> info) {
-		int i = x >> 2;
-		int j = z >> 2;
-		if ((long) i * (long) i + (long) j * (long) j <= 4096L) {
-			info.setReturnValue(this.theEndBiome);
-		} else {
-			float noise = EndBiomeProvider.getRandomNoise(this.generator, i * 2 + 1, j * 2 + 1);
+		if (
+				info.getReturnValue().equals(endHighlandsBiome) ||
+						info.getReturnValue().equals(endMidlandsBiome) ||
+						info.getReturnValue().equals(endBarrensBiome)
+		) {
 			Biome biome = getNoiseBiome(x, z);
-			boolean isChorus = biome == this.endMidlandsBiome;
-			if (noise > 40.0F) {
-				info.setReturnValue(isChorus ? this.endHighlandsBiome : biome);
-			} else if (noise >= 0.0F) {
-				info.setReturnValue(isChorus ? this.endMidlandsBiome : biome);
-			} else {
-				info.setReturnValue(noise < -20.0F ? this.smallEndIslandsBiome : isChorus ? this.endBarrensBiome : biome);
+			boolean isChorus = biome == lookupRegistry.getOrDefault(new ResourceLocation("minecraft:end_midlands"));
+			if (!isChorus) {
+				info.setReturnValue(biome);
```

Comment:
Why lookup the `Biome` for `end_midlands` every time?
The `Biome` for `end_midlands` gets stored in this provider as `endMidlandsBiome`
The `isChorus` local variable is redundant now because it gets used once. Also, why declare a local variable only to flip its value when testing it later?
```suggestion
			if (biome != this.endMidlandsBiome) {
				info.setReturnValue(biome);
			}
```